python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -561,24 +583,104 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         )
 
 
+@whitelist_for_serdes
+@experimental
+class AutoMaterializeAssetPartitionFilter(
+    NamedTuple(
+        "_AutoMaterializeAssetPartitionFilter",
+        [("latest_run_required_tags", Optional[Mapping[str, str]])],
+    )
+):
+    """A filter that can be applied to an asset partition, during auto-materialize evaluation, and
```

Comment:
in reality this is applied to a set of asset partitions, not just one

Reply:
From the perspective of the person constructing the class, I think it's easiest to think it about as a per-asset partition filter. The fact that it works in batch is an implementation detail. Open to pushback on this of course.

Reply:
Ok actually I'm realizing that both `RunsFilter` and `EventRecordsFilter` are plural, so will change to be consistent.